extensions/shared/README.md

```diff
@@ -1,3 +1,19 @@
 # Jetpack Block Editor Extensions: Shared
 
 The shared directory is meant to contain components, scripts, and stylesheets that are used in multiple blocks.
+
+## getJetpackData()
+
+Gets a list of available blocks, and jetpack's connection status.
+On a Jetpack site, there are special cases when a block is considered unavailable.
+For example, the site may not have the required module, they may not have the required plan,
+or the site admin may have filtered out the use of a certain block.
+
+## registerJetpackBlock()
+
+This util will only register a block if it meets the availability requirements described above.
+
+
+## registerJetpackPlugin()
+
+This util will only register a Gutenberg plugin if it meets the availability requirements described above.
```

Comment:
We probably need to tell about these methods in more visible `/extensions/README.md` as part of documentation to get folks started building blocks.